dds/DCPS/MultiTopicDataReader_T.cpp

```diff
@@ -104,33 +102,33 @@ MultiTopicDataReader_T<Sample, TypedDataReader>::join(
       SampleInfo info;
       const ReturnCode_t ret = other_dri->read_instance_generic(other_data.ptr_,
         info, ih, READ_SAMPLE_STATE, ANY_VIEW_STATE, ALIVE_INSTANCE_STATE);
-      if (ret != RETCODE_OK && ret != RETCODE_NO_DATA) {
+      if (ret == DDS::RETCODE_NO_DATA || !info.valid_data) {
+        return DDS::RETCODE_NO_DATA;
+      }
+
+      if (ret != RETCODE_OK) {
         throw std::runtime_error(
-         OPENDDS_STRING("In join(), incoming DataReader for ") + OPENDDS_STRING(other_topic) +
-         " read_instance_generic: " + retcode_to_string(ret));
-      } else if (ret == DDS::RETCODE_OK) {
+          OPENDDS_STRING("In join(), incoming DataReader for ") + OPENDDS_STRING(other_topic) +
```

Comment:
Do we really want to `throw` here?  What about returning the error?

Reply:
I'm not sure why exception was used in the first place, but probably it could just use the return code.